### PR TITLE
Redirect user to concierge when onboarding completed

### PR DESCRIPTION
**Pull request by skyweb331**
### Details
Redirect where user was when they try to deeplink to the onboarding flow after completion

### Fixed Issues
$ https://github.com/Expensify/App/issues/40876
PROPOSAL: https://github.com/Expensify/App/issues/40876#issuecomment-2081298473

### Tests
1. Complete the onboarding steps for current login.
2. Try to visit onboarding url, https://dev.new.expensify.com:8082/onboarding, via deeplink or directly
3. Should be redirected to concierge page.

- [x] Verify that no errors appear in the JS console

### Offline tests
If user already completed onboarding steps, https://staging.new.expensify.com/onboarding should be redirected to concierge page.
If not, this will not work, because onboarding completion status comes from Backend.

### QA Steps
1. Complete the onboarding steps for current login.
2. Try to visit onboarding url, https://staging.new.expensify.com/onboarding, via deeplink or directly
3. Should be redirected to concierge page.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/25519849/5ebabc98-c006-41da-8df8-4408426572c9

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/25519849/e01a9578-627c-436c-bf3f-6214d37d5505

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/25519849/2d60d7f5-a870-4837-870a-8e97e692e0ac

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/25519849/7dc243f0-0514-4c2a-9a82-d6e2e8eccad0

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/25519849/174e0660-6d34-4c7c-956a-7e510f7956c6

</details>

### CODE REVIEW

**Comment by skyweb331** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by c3024** [comment]
The modal appears and after loading is complete it disappears.

https://github.com/Expensify/App/assets/102477862/2b122c3d-15f2-4d35-b608-a3d1298fa489

Can we change this to not show the modal at all for already onboarded users?

**Comment by skyweb331** [comment]
@c3024 I will look into this. BTW, this behavior depends on `nvp_onboarding.hasCompletedGuidedSetupFlow` so if current user session still in receiving this state, modal will be visible...

To solve this,
1. Set  `nvp_onboarding.hasCompletedGuidedSetupFlow` as `true` optimistically ( as most current users are already completed onboarding), so every users will be redirected to concierge by default.
2. For non-completed users, once BE sends `false` for this value, I will redirect the user to onboarding page.

Let me know your thoughts on this.

**Comment by c3024** [comment]
I agree. 

We are showing the modal only after this promise is resolved here as well so it should be fine.

https://github.com/Expensify/App/blob/cba6916458c91e26ab8fb795956df0aa9508d6ab/src/libs/actions/Welcome.ts#L33

**Comment by skyweb331** [comment]
@c3024 I set onboarding status completed as initial.

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (c3024) [inline src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:64]: `useOnyx` is preferred to `withOnyx` HOC. Could you change it?

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (c3024) [inline src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:34]:
```suggestion
        return null;
```

File: src/types/onyx/index.ts
Comment (c3024) [inline src/types/onyx/index.ts:112]: What is this for? I don't see this export from this file being used anywhere.

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (c3024) [inline src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:32]:
```suggestion
```

I think just going back without redirecting to Concierge is better. Let me know if there are any edge cases in which redirecting to Concierge is useful.

**Review by c3024 — CHANGES_REQUESTED** [review]

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (skyweb331) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:32]: Suppose you are in a report and clicked `https://dev.new.expensify.com:8082/onboarding` link in a message. In this case, if there's only `goBack`,  it will redirect user to onboarding first and then will go back to the current report. So need to redirect user to Concierge again.

What if there's only one `Concierge` redirect? If so, user will be redirect to concierge on deep link click, but if he clicks go back in the browser, it will redirect user to onbarding and then onboarding will redirect user back to concierge, so user can not go back... so these two actions are required and it will remove onboarding from route history...

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (skyweb331) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:34]: For me, when I test it on IOS and android, `return null` throws some errors. That's why I put Fragment. Need to test again though.

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (c3024) [inline src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:31]:
```suggestion
    useEffect(() => {
        if (hasCompletedGuidedSetupFlow) {
            Navigation.goBack();
            Report.navigateToConciergeChat();
        }
    }, [hasCompletedGuidedSetupFlow]);

    if (hasCompletedGuidedSetupFlow) {
        return null;
    }
```

Is this the error you found in native apps?
> ERROR  Warning: Cannot update a component (`ForwardRef(BaseNavigationContainer)`) while rendering a different component (`OnboardingModalNavigator`). To locate the bad setState() call inside `OnboardingModalNavigator`, follow the stack trace as described in https://react.dev/link/setstate-in-render

I think this is due to using navigation while rendering. So we need to move it to a `useEffect`.

**Comment by skyweb331** [comment]
@c3024 Thanks for your professional review. Just pushed the fixes as you guided.

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (c3024) [inline src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:34]:
```suggestion
        Navigation.isNavigationReady().then(() => {
            Navigation.goBack();
            Report.navigateToConciergeChat();
        }
```

https://github.com/Expensify/App/assets/102477862/20c811c0-5999-4ace-b257-375c7946219b

I see the navigator is not ready when navigating to these routes. So, it goes back to onboarding again as seen in the video.

Comment (skyweb331) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx]: I thought, this component is defined in Route, so navigation should be ready already...Pushed fix as you mentioned

**Comment by c3024** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by c3024 — APPROVED** [review]
LGTM

File: src/ONYXKEYS.ts
Comment (mountiny) [inline src/ONYXKEYS.ts:581]: Why did we change the type? for old accounts the nvp is an empty array so I think we need to keep this

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:32]:
```suggestion
            // otherwise going back on Concierge will go to onboarding and then redirected to Concierge again
```

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:24]: I see you have changed it here to work with the type change, however, is removing the empty array type compatible with all the other places where we use this nvp?

**Review by mountiny — COMMENTED** [review]
This works for me, but we need to make sure all the other places in the Ap pare updated to work with removing the [] empty array style

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (skyweb331) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:24]: @mountiny 
I checked every place where `ONYXKEYS.NVP_ONBOARDING` is used in app and it does not need to be an array. Even though Backend returns empty array for old accounts, `[].hasCompletedGuidedSetupFlow` is `undefined` so this should work fine.

Let me know your thoughts on this.
Thanks.

Comment (mountiny) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:24]: @skyweb331 I am still not sure if we should remove the empty array type, given this is a valid and expected value for that nvp, I am worried that if people will forget it can be empty array for old accounts, they might make some changes that wont treat the case safely.

Could you please try to put that type back? thanks

Comment (skyweb331) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx]: @mountiny I reverted it. And also added comment why it could be an array.

**Review by mountiny — DISMISSED** [review]
NAB @c3024 Can you please do a quick test on web once more now after the changes?

**Comment by mountiny** [comment]
Lint is failing too @skyweb331

**Comment by c3024** [comment]
Tests well

https://github.com/Expensify/App/assets/102477862/e84b57b7-579c-447f-a6da-507bf5293e21


https://github.com/Expensify/App/assets/102477862/39c357d4-35fe-4e91-906b-fba9c5331269


This is the difference `prettier` is showing that is causing the lint failure. I don't see any difference. 🫤 

Screenshot 2024-05-24 at 8 22 25 PM

**Comment by c3024** [comment]
@skyweb331 

please fix lint by running `npm run prettier`

**Comment by skyweb331** [comment]
@c3024 Prettier fixed.

**Review by c3024 — APPROVED** [review]
LGTM

**Review by mountiny — DISMISSED** [review]
Thanks!

**Comment by mountiny** [comment]
@skyweb331 sorry, can you please resolve merge conflicts before I can merge?

**Comment by skyweb331** [comment]
@mountiny Done.

**Comment by c3024** [comment]
@skyweb331 

Please move this

```js
    if (hasCompletedGuidedSetupFlow) {
        return null;
    }
```
 in `OnboardingModalNavigator` below the newly added hooks.

Hooks should not be added below `null` return.

Lint is failing. 

There is a prettier difference as well.

Please run `npm run prettier` and `npm run lint` and fix errors (if any) every time before committing.

**Review by mountiny — DISMISSED** [review]
Thanks!

**Comment by mountiny** [comment]
@skyweb331 please address the feedback whenever you can, thanks!

**Comment by skyweb331** [comment]
@c3024 Sorry for wrong merging. I will run `npm run prettier && npm run lint` every time.

**Review by c3024 — CHANGES_REQUESTED** [review]

**Comment by skyweb331** [comment]
@c3024 Done.

**Review by mountiny — DISMISSED** [review]
Thanks

**Comment by mountiny** [comment]
@c3024 can you please approve if all looks good for you. I think the comments were addressed but nowhere to rush

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (c3024) [inline src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:52]:
```suggestion
    if (hasCompletedGuidedSetupFlow) {
        return null;
```

@skyweb331 
Why did you change to negation here? 

Why is `null` omitted here? Returning `undefined` is not allowed for React components. You should explicitly return `null`.

**Review by c3024 — CHANGES_REQUESTED** [review]

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (skyweb331) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx]: @c3024 Done. Thanks for your great point out.

**Review by c3024 — APPROVED** [review]
Tests well. LGTM!


https://github.com/Expensify/App/assets/102477862/e6f8a323-f05a-466e-8417-4d3002741214

**Review by mountiny — APPROVED** [review]
thanks

File: src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx
Comment (skyweb331) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:28]: You mean, empty array means `onboarding` completed?

Comment (mountiny) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:28]: Yeah, thats a mistake. When the hasCompleted property is not defined it means the onboarding was completed given its old user and we do not want to show them the flow in that case

Comment (skyweb331) [reply src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx:28]: As far as I remember, `nvp_onbarding` was empty array for new users in NewDot, which means, array indicates onboarding is not completed...
Is there any backend changes for this?

**Pull request merged**
